libcodechecker/server/api/report_server.py

```diff
@@ -1865,180 +1949,33 @@ def massStoreRun(self, name, tag, version, b64zip, force):
-                            last_report_event['location']['line'],
-                            bug_id,
-                            report.main['check_name'])
-
-                        supp = sp_handler.get_suppressed()
-                        if supp:
-                            _, _, comment = supp
-                            status = ttypes.ReviewStatus.FALSE_POSITIVE
-                            self._setReviewStatus(report_id, status, comment,
-                                                  session)
-
-                    LOG.debug("Storing done for report " + str(report_id))
-
-            reports_to_delete = set()
-            for bug_hash, reports in hash_map_reports.items():
-                if bug_hash in new_bug_hashes:
-                    reports_to_delete.update(map(lambda x: x.id, reports))
-                else:
-                    for report in reports:
-                        report.detection_status = 'resolved'
-                        report.fixed_at = run_history_time
-
-            if len(reports_to_delete) != 0:
-                session.query(Report) \
-                    .filter(Report.id.in_(reports_to_delete)) \
-                    .delete(synchronize_session=False)
-
+            durations = 0
             if len(check_durations) > 0:
-                store_handler.setRunDuration(self.__storage_session,
+                # Round the duration to seconds.
+                durations = int(sum(check_durations))
+
+            with DBSession(self.__Session) as session:
+                run_id = store_handler.addCheckerRun(session,
+                                                     command,
+                                                     name,
+                                                     tag,
+                                                     user,
+                                                     run_history_time,
+                                                     version,
+                                                     force)
+                self.__store_reports(session,
+                                     report_dir,
+                                     source_root,
+                                     run_id,
+                                     file_path_to_id,
+                                     run_history_time)
+
+                store_handler.setRunDuration(session,
                                              run_id,
-                                             # Round the duration to seconds.
-                                             int(sum(check_durations)))
+                                             durations)
 
-            # The database session initialized in addCheckerRun should be
-            # closed in finishCheckerRun after the report storage is done.
-            store_handler.finishCheckerRun(self.__storage_session, run_id)
+                store_handler.finishCheckerRun(session, run_id)
+
+                session.commit()
```

Comment:
This is definitely the only part we `commit`, yes?

Reply:
The massStore is split up to two main store sections, at first the source files are stored (each file in a new session). Everything after the source files are stored is done in one session (storing the reports), it should be successful and committed once or it should fail and rollback.